sql: fix a bunch of minor spacing issues in the grammar

BramGruneir opened the pull request:
I was trying to do some searching and couldn't find what I was looking for due
to inconsistent spacing.

We should move towards adding a linter for the grammer.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/25216)
<!-- Reviewable:end -->

knz commented:
How did you do this change? Did you automate it? How?

BramGruneir commented:
If I automated it I would have included the steps.  

Finding the double, triple, quadruple spaces was easy.  Just a search for `  ` and multi-line edit to fix them.

The rest was looking for `: ` and see if things didn't line up.  So that was more manually, but I needed a break from other work and just did it on autopilot while enjoying some music.  Didn't take very long.

knz reviewed (commented):
This patch is a mixed bag of desirable stuff (the alignment of %type directives) and less desirable stuff. I wish we could break this up in multiple parts so we can discuss them separately.

I am also a little bit concerned that any subsequent patches on the 2.0 branch will get merge conflicts.

BramGruneir commented on pkg/sql/parser/sql.y:
```diff
@@ -433,109 +433,109 @@ func newNameFromStr(s string) *tree.Name {
 
-%token <str>   ZONE
+%token <str> ZONE
 
```
That's not entirely true.

`%token <*tree.NumVal> ICONST FCONST` uses only one space already.

I just don't know of any instance in which we use a double spaces for

knz replied:
ok let's keep your change here.

BramGruneir commented on pkg/sql/parser/sql.y:
```diff
@@ -545,13 +545,13 @@ func newNameFromStr(s string) *tree.Name {
+%token NOT_LA WITH_LA AS_LA
 
 %union {
-  id             int
```
This one was weird.  I don't understand the huge spacing.  I can revert it.

knz replied:
let's keep it your way.

BramGruneir commented on pkg/sql/parser/sql.y:
```diff
@@ -1372,19 +1372,19 @@ alter_table_cmd:
   {
     $$.val = &tree.AlterTableDropColumn{
       ColumnKeyword: $2.bool(),
-      IfExists: true,
```
More than half of these were already aligned as if it were using gofmt.  So that's why I made the change.  We should have a consistent style and we don't.  I  opted for the style that matches the rest of our codebase.

Manually adding the spaces ins't a big deal but without linter support, it will not be enforceable.

There IS yacc file support in a lot of editors, but our specific blend of go and the grammar broke every one that I tried.

knz replied:
> More than half of these were already aligned as if it were using gofmt. 

I don't see that's true. Can you show me?

BramGruneir replied:
Just went back through and did a bit of an audit of them.  Most were not formatted.  But there is a mix.

Take a look at the `$$.val = &tree.CreateIndex{` Of the 4, 2 were formatted, 1 was not, and 1 was half formatted.

I'd still rather we format them in gofmt style to match the rest of our codebase.  But I'll leave the final decision up to you.  I find it more readable and I don't think the manual alignment is hard to do or enforce.  But the most important thing here is to be consistent.  So let me know either way.

knz reviewed (commented):
I am coming to like most of the changes. I still have reservations about the struct initializations. Please provide more information (see comment below).

knz commented:
Let's skip the struct alignment for now unless you also can implement a linter for it straight away. I don't see anyone in the team willing (or able) to properly check this manually during future reviews.

The rest LGTM

---

Reviewed 1 of 1 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/25216#-:-LCV8EKf93tVN3xi4fR1:b-bnj2qf)*
<!-- Sent from Reviewable.io -->

BramGruneir committed:
sql: fix a bunch of minor spacing issues in the grammar

I was trying to do some searching and couldn't find what I was looking for due
to inconsistent spacing.

We should move towards adding a linter for the grammer.

Release note: None

BramGruneir commented:
Done.  I've created a branch with just the spacing changes for for the struct alignment, so we don't have to start from scratch if we do move in that direction.

https://github.com/BramGruneir/cockroach/tree/spacing2

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/25216#-:-LCZwmHQ7-MZU8Xu3kvL:bmgb6jh)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
bors r+

knz commented:
Thanks!

BramGruneir commented:
bors r+

BramGruneir commented:
bors r+

On Tue, May 15, 2018, 16:57 craig[bot] <notifications@github.com> wrote:

> Build failed (retrying...)
>
>    - GitHub CI (Cockroach)
>    <https://teamcity.cockroachdb.com/viewLog.html?buildId=659931&buildTypeId=Cockroach_UnitTests>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/25216#issuecomment-389310868>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABihuZexSEgDaORPwKpWfZe2bj2VPy4Bks5ty0EugaJpZM4TuVnL>
> .
>
